src/client.c

```diff
@@ -198,14 +198,14 @@ int raft_add(struct raft *r,
 
     req->cb = cb;
 
+    assert(r->leader_state.change == NULL);
+    r->leader_state.change = req;
```

Comment:
You should reset `r->leader_state.change` to NULL if `clientChangeConfiguration` fails, same for the other similar changes.

Reply:
Yep, I think it will be perfect)